contrib/IECoreAlembic/src/IECoreAlembic/MeshWriter.cpp

```diff
@@ -152,31 +152,32 @@ class MeshWriter : public PrimitiveWriter
 				);
 			}
 
-			const FloatVectorData *sData = meshPrimitive->variableData<FloatVectorData>( "s" );
-			const FloatVectorData *tData = meshPrimitive->variableData<FloatVectorData>( "t" );
-			vector<V2f> st;
-			if( sData && tData )
+			std::vector<unsigned int> indices;
+			PrimitiveVariableMap::const_iterator uvIt = meshPrimitive->variables.find( "uv" );
+			if( uvIt != meshPrimitive->variables.end() )
 			{
-				const vector<float> &s = sData->readable();
-				const vector<float> &t = tData->readable();
-				if( s.size() == t.size() )
+				const V2fVectorData *uvData = runTimeCast<V2fVectorData>( uvIt->second.data.get() );
+				if( uvData )
 				{
-					st.reserve( s.size() );
-					for( vector<float>::const_iterator sIt = s.begin(), tIt = t.begin(), eIt = s.end(); sIt != eIt; ++sIt, ++tIt )
+					OV2fGeomParam::Sample uvSample( uvData->readable(), geometryScope( uvIt->second.interpolation ) );
+
+					if( uvIt->second.indices )
 					{
-						st.push_back( V2f( *sIt, *tIt ) );
+						const std::vector<int> &indexValues = uvIt->second.indices->readable();
```

Comment:
Hmm, I wonder, should we be storing the indices as unsigned anyway? What's the most common approach? If we do need the conversion, maybe `indexValues( indices.begin(), indices.end() )` would be a more concise way of constructing it?

Reply:
Yeah, I stumbled when I hit this one. But we don't store vertexIds and unsigned, so I think either we change both (in another PR) or we leave it.

Reply:
The trouble with constructing from iterators is that the indices array needs to remain alive until `writeArbGeomParams` has been called or I get hard crashes. That's why I lifted its construction outside the conditional. I guess I could use `std::copy`? I think I tried that and got crashes as well, but that may have been before I realized the lifetime thing...

Reply:
> Yeah, I stumbled when I hit this one. But we don't store vertexIds as unsigned, so I think either we change both (in another PR) or we leave it.

Yeah, that's RenderMan legacy I think - that defines everything with ints even if its gotta be positive, and quite a lot of things have followed suit. If we convert indices to unsigned then we'll have fewer conversions to do, but if we make vertexIds unsigned I think we'll be doing more conversions probably. Just wanted to bring it up, but we don't have to deal with it now.

> The trouble with constructing from iterators is that the indices array needs to remain alive until writeArbGeomParams has been called or I get hard crashes. That's why I lifted its construction outside the conditional. I guess I could use std::copy?

Ah, right. Yep, I think that's caught me out before as well. Could just do `indices = vector<unsigned int>( indexValues.begin(), indexValues.end() )`? I _think_ that'll do a move assignment because of C++11 fanciness, although I'd want to check. Or, more old school, `indices.swap( vector<unsigned int>( indexValues.begin(), indexValues.end() ) )`?

Reply:
Addressed in 8843fc72ae5d1bca461a54f5c65e55bebd86491b